File: src/main/java/com/ject/studytrip/studylog/application/dto/StudyLogDetail.java
package com.ject.studytrip.studylog.application.dto;

import com.ject.studytrip.studylog.domain.model.StudyLog;
import com.ject.studytrip.studylog.domain.model.StudyLogDailyMission;
import java.util.Collections;
import java.util.List;
import java.util.Optional;

public record StudyLogDetail(
        StudyLogInfo studyLogInfo, List<StudyLogDailyMissionInfo> studyLogDailyMissionInfos) {
    public static StudyLogDetail from(
            StudyLog studyLog, List<StudyLogDailyMission> studyLogDailyMissions) {
        List<StudyLogDailyMission> safeStudyLogDailyMissions =
                Optional.ofNullable(studyLogDailyMissions).orElse(Collections.emptyList());

        return new StudyLogDetail(
                StudyLogInfo.from(studyLog),
                safeStudyLogDailyMissions.stream().map(StudyLogDailyMissionInfo::from).toList());
    }
}

File: src/main/kotlin/com/ject/studytrip/studylog/application/dto/StudyLogDetail.kt
package com.ject.studytrip.studylog.application.dto

import com.ject.studytrip.studylog.domain.model.StudyLog
import com.ject.studytrip.studylog.domain.model.StudyLogDailyMission

data class StudyLogDetail(
    val studyLogInfo: StudyLogInfo,
    val studyLogDailyMissionInfos: List<StudyLogDailyMissionInfo>,
) {
    companion object {
        fun from(
            studyLog: StudyLog,
            studyLogDailyMissions: List<StudyLogDailyMission>?,
        ): StudyLogDetail {
            val studyLogDailyMissions = studyLogDailyMissions ?: emptyList()

            return StudyLogDetail(
                StudyLogInfo.from(studyLog),
                studyLogDailyMissions.map { StudyLogDailyMissionInfo.from(it) },
            )
        }
    }
}
